src/forms/compoundField.js

```diff
@@ -0,0 +1,153 @@
+const Joi = require('joi');
+const option = require('option');
+const FieldError = require('./fieldError');
+
+class BadValidationTargetError extends Error {
```

Comment:
This reads kinda like the issue is the target. In my head it makes me thing "this is the error for what happens when the validator fails to mount on the correct target" rather than what it is.

Reply:
Right, WrongValidationTarget? or ValidationTargetNotFound?

Reply:
went for `TargetNotFoundError`. Done